### PR TITLE
Added form level validators

### PR DESCRIPTION
**Pull request by claytn**
### Added: 
* Validation for inputs in application form

### Libraries Used:
* [redux-form-validators](https://www.npmjs.com/package/redux-form-validators#date)
* [react-intl](https://github.com/yahoo/react-intl) (dependency for redux-form-validators)

### CODE REVIEW

File: src/components/apply/formValidators.js [hunk 1]
```diff
@@ -1,27 +1,62 @@
...
+  city: [requiredInput],
+  state:[requiredInput],
+  zipCode:[requiredInput, numberValidationWithLength({length: 5})]
+};
+
+// Applies all the validator functions to the input, returning the value of the first rejection
+// or `undefined`.
+const applyValidators = (validators, value, values) => _.find(_.map(validators, validator => validator(value, values)));
+
+/* This function is adapted from code from redux-form-validators
```
Comment (jimmyhmiller) [inline src/components/apply/formValidators.js:45]: What is the corresponding function in `redux-form-validators`? Why doesn't it work in our case?

File: src/index.js [hunk 2]
```diff
@@ -29,7 +30,9 @@ const Root = () => {
   return (
     <MuiThemeProvider>
       <Provider store={store}>
-        <ApplicationPage history={history} />
+        <IntlProvider locale="en">
```
Comment (jimmyhmiller) [inline src/index.js:33]: Any reason why this was added?

File: src/components/apply/formValidators.js [hunk 3]
```diff
@@ -1,27 +1,62 @@
...
-  const errors = {}
-  for (let field in validations) {
-    let value = values[field]
-    errors[field] = validations[field].map(validateField => {
-      return validateField(value, values)
-    }).find(x => x)
+const dobValidation = addValidator({
+  defaultMessage: validationMsg('Must be 18 years or older.'),
+  validator: (options, value) => {
+    let ageMs = Date.now() - new Date(value).getTime();
```
Comment (jimmyhmiller) [inline src/components/apply/formValidators.js]: Why `let`?

File: src/components/apply/formValidators.js [hunk 4]
```diff
@@ -1,27 +1,62 @@
...
+    return (new Date(ageMs).getUTCFullYear() - 1970) >= 18;
```
Comment (jimmyhmiller) [inline src/components/apply/formValidators.js]: Might be worth looking at [moment](http://momentjs.com/).

**Review by jimmyhmiller — CHANGES_REQUESTED** [review]

File: src/components/apply/formValidators.js (hunk 1 above)
Comment (Ameobea) [reply src/components/apply/formValidators.js:45]: They don't have a built-in supplied approach, but included a for-loop example in their documentation:

```
validate = (values) => {
  const errors = {}
  for (let field in validations) {
    let value = values[field]
    errors[field] = validations[field].map(validateField => {
      return validateField(value, values)
    }).find(x => x)
  }
  return errors
}
```

This has the same functionality but is implemented a bit more in-line with the conventions of this project.

File: src/index.js (hunk 2 above)
Comment (claytn) [reply src/index.js:33]: It is required by redux-form-validators.  React-Intl is used for creating FormattedMessages which are used as defaultMessages by the redux-form-validators.

File: src/components/apply/formValidators.js (hunk 3 above)
Comment (claytn) [reply src/components/apply/formValidators.js]: Fixed.

File: src/components/apply/formValidators.js (hunk 4 above)
Comment (claytn) [reply src/components/apply/formValidators.js]: Added.

File: src/components/apply/formValidators.js (hunk 1 above)
Comment (jimmyhmiller) [reply src/components/apply/formValidators.js:45]: That's weird that it isn't built in. Definitely prefer the version you have now to the one posted. Might be worthwhile to `_.mapValues` and see if you can make an even simpler version.

File: src/index.js (hunk 2 above)
Comment (jimmyhmiller) [reply src/index.js:33]: Guess I should have read that in the first comment.

**Review by jimmyhmiller — APPROVED** [review]

**Pull request merged**